lib/core/src/sync/model/mod.rs

```diff
@@ -0,0 +1,68 @@
+use boltz_client::boltz::SwapTree;
+use serde::{Deserialize, Serialize};
+
+use crate::prelude::Direction;
+
+pub(crate) mod sync;
+
+#[derive(Serialize, Deserialize)]
+pub(crate) struct ClaimDetails {
+    pub(crate) swap_tree: SwapTree,
+    pub(crate) timeout_block_height: u32,
+    pub(crate) claim_fees_sat: u64,
+    pub(crate) claim_address: String,
+    pub(crate) claim_private_key: String,
+}
+
+#[derive(Serialize, Deserialize)]
+pub(crate) struct LockupDetails {
+    pub(crate) swap_tree: SwapTree,
+    pub(crate) timeout_block_height: u32,
+    pub(crate) lockup_address: String,
+    pub(crate) refund_private_key: String,
+}
+
+#[derive(Serialize, Deserialize)]
+pub(crate) struct ChainSyncData {
+    pub(crate) swap_id: String,
+    pub(crate) preimage: String,
+    pub(crate) description: Option<String>,
+    pub(crate) direction: Direction,
+    pub(crate) claim_details: ClaimDetails,
```

Comment:
I prefer to not use nested structures here if possible. It makes it more clear to see what data we sync.